memcnn/models/tests/test_split_dim.py

```diff
@@ -0,0 +1,29 @@
+from typing import Union
+
+import pytest
+import torch
+
+from memcnn import AdditiveCoupling, AffineCoupling, InvertibleModuleWrapper
+
+
+class Check(torch.nn.Module):
+    def __init__(self, dim, target_size):
+        super(Check, self).__init__()
+        self.dim = dim
+        self.target_size = target_size
+
+    def forward(self, fn_input: torch.Tensor) -> torch.Tensor:
+        assert fn_input.size(self.dim) == self.target_size
+        return fn_input
+
+
+@pytest.mark.parametrize('dimension', [None, 1, 2])
```

Comment:
You might also want to test the batch dimension since this should be possible as well.

```suggestion
@pytest.mark.parametrize('dimension', [None, 0, 1, 2])
```

Reply:
I explicitly removed it, as it'd kill the affine coupling. If you can use it with 1 and 2, 0 should work fine (with everything other than affine coupling) as well.

Reply:
Interesting, I'll check tomorrow why that's happening.

Reply:
Affine coupling splits the output of the function in two tensors for its multiplication. If we first split it in two and then split it in two different chunks each, it can't work.

Reply:
Alright, I see what's going on here. The affine coupling expects the module to output two tensors instead of one. The best solution here is to set the adapter property to the AffineAdapterNaive class when creating the AffineCoupling class. Here are some proposed changes that should make it work (I have tested these changes on my machine):

Add AffineAdapterNaive to the imports from line 5:
```python
from memcnn import AdditiveCoupling, AffineAdapterNaive, AffineCoupling, InvertibleModuleWrapper
```

And modify the test lines 19-end with:
```python
@pytest.mark.parametrize('dimension', [None, 0, 1, 2])
@pytest.mark.parametrize('coupling', [AdditiveCoupling, AffineCoupling])
def test_split_dim(dimension, coupling):
    input_size = (4, 8, 12)
    dim = 1 if dimension is None else dimension
    module = Check(dim, input_size[dim] // 2)
    coupling_args = dict(adapter=AffineAdapterNaive) if coupling.__name__ == 'AffineCoupling' else dict()
    if dimension is not None:
        coupling_args["split_dim"] = dimension
    model = coupling(module, **coupling_args)
    model = InvertibleModuleWrapper(model)
    inp = torch.randn(input_size, requires_grad=False)
    output = model(inp)
    assert inp.shape == output.shape

```